packages/core/src/module-resolver.ts

```diff
@@ -458,27 +458,27 @@ export class Resolver {
 
     // first, the various places our template might be.
     for (let candidate of this.componentTemplateCandidates(target.packageName)) {
+      let candidateSpecifier = `${target.packageName}${candidate.prefix}${target.memberName}${candidate.suffix}`;
       let resolution = this.nodeResolve(
         `${target.packageName}${candidate.prefix}${target.memberName}${candidate.suffix}`,
         target.from
       );
       if (resolution.type === 'real') {
-        hbsModule = resolution.filename;
+        hbsModule = candidateSpecifier;
```

Comment:
This is similar to what we did in https://github.com/embroider-build/embroider/pull/1648. Both are cases where we were rewriting requests to absolute paths, which ends up looking to vite like the user actually typed an absolute path in their import, which gets different behavior than a normal relative- or package-name import.

In this case when we successfully resolve a specifier to a component file to check that it exists, rather than rewriting the request to point directly at that file we rewrite the request to use the specifier that we just successfully resolved.

(This duplicate resolving is only needed when supporting the ambiguity of loose-mode templates. As people migrate to template tag they stop needing it.)